### ISSUE TITLE
[ISSUE] Error: unexpected error handling request: json: cannot unmarshal number into Go struct field APIErrorBody.error_code of type string.

### ISSUE THREAD

**Issue by hengrumay** [databricks/databricks-sdk-go#1060]
**Description**

Run databricks bundle validate -t mmt_awsgha_mlops_test
Error: unexpected error handling request: json: cannot unmarshal number into Go struct field APIErrorBody.error_code of type string. This is likely a bug in the Databricks SDK for Go or the underlying REST API. Please report this issue with the following debugging information to the SDK issue tracker at https://github.com/databricks/databricks-sdk-go/issues. Request log:
```
GET /api/2.0/preview/scim/v2/Me
> * Host: 
> * Accept: application/json
> * Authorization: REDACTED
> * Traceparent: 00-[1](https://github.com/hengrumay/awsgha_mlops_demo/actions/runs/11231672149/job/31221671542?pr=1#step:4:1)779b642020b0a5a4e30ecb4c5ab7c44-ba1d7d3ec4ec7624-01
> * User-Agent: cli/0.221.0 databricks-sdk-go/0.42.0 go/1.21.10 os/linux cmd/bundle_validate auth/pat cicd/github
< HTTP/2.0 403 Forbidden
< * Content-Length: 127
< * Content-Type: application/json; charset=utf-8
< * Date: Tue, 08 Oct 2024 08:3[6](https://github.com/hengrumay/awsgha_mlops_demo/actions/runs/11231672149/job/31221671542?pr=1#step:4:7):53 GMT
< * Server: databricks
< * Vary: Accept-Encoding
< * X-Databricks-Reason-Phrase: Source IP address: 20.168.76.149 is blocked by Databricks IP ACL for workspace: 1444828305810485
< * X-Request-Id: dbeb6518-77eb-4f7a-b7e4-c68191ad2b81
< {
<   "error_code": 403,
<   "message": "Source IP address: 20.168.76.149 is blocked by Databricks IP ACL for workspace: 1444[8](https://github.com/hengrumay/awsgha_mlops_demo/actions/runs/11231672149/job/31221671542?pr=1#step:4:9)28305810485"
< }
```
Error: Process completed with exit code 1.

Also checked that the IP is in allow list -- So there's a dynamic github IP that is not in the AWS/workspace allow list and we need to have a dedicated runner or a way to programmatically include the dynamically generated IP to be added to the allow list (and remove it when done or we will have. a lot of allowed IPs)
```
may.merkletan@MTXRM6N94W awsgha_mlops_demo % databricks  ip-access-lists create --json '{
 "label": "###",
 "list_type": "ALLOW",
 "ip_addresses": [
   "##.###.##.###"  // this gets dynamically created but is not part of the allow-list 
  ]
}'
```

**Other Information**
 - OS: MAC M2 Pro 
 - Version: [14.6.1]

**Additional context**
Trying to set up MLOps Stacks using Service Principal on [e2-field](https://e2-demo-field-eng.cloud.databricks.com/?o=1444828305810485)
Able to run directly with databricks cli 
```
may.merkletan@MTXRM6N94W awsgha_mlops_demo % databricks bundle deploy -t mmt_awsgha_mlops_test --profile DEFAULT
Uploading bundle files to /Users/may.merkletan@databricks.com/.bundle/awsgha_mlops_demo/mmt_awsgha_mlops_test/files...
Deploying resources...
Updating deployment state...
Deployment complete!
```

**Comment by hengrumay** [comment]
it seems that the dynamic IP allocation isn't part of the workspace allow list  -- is there a way to have a dedicated runner IP for each instantiated repo? 
<img width="1572" alt="image" src="https://github.com/user-attachments/assets/8d3d33f6-4bfc-4925-a49b-db7839a7a222">
<img width="939" alt="image" src="https://github.com/user-attachments/assets/c60d7392-a337-417c-8428-c05645a27be7">

**Comment by hengrumay** [comment]
similar issues observed previously https://github.com/databricks/databricks-sdk-go/issues/871#issuecomment-2085546315

**Comment by andrewnester** [comment]
I'm going to close this issue because it's not related to Go SDK or CLI but Github runners and Databricks workspace allow listing
